pool/paymentmgr.go

```diff
@@ -488,108 +483,63 @@ func (pm *PaymentMgr) applyTxFees(inputs []chainjson.TransactionInput, outputs m
 	return sansFees, estFee, nil
 }
 
-// fetchTxConfNotifications is a helper function for fetching tx confirmation
-// notifications. It will return when either a notification or error is
-// received from the provided notification source, or when the provided
-// context is cancelled.
-func fetchTxConfNotifications(ctx context.Context, notifSource func() (*walletrpc.ConfirmationNotificationsResponse, error)) (*walletrpc.ConfirmationNotificationsResponse, error) {
-	funcName := "fetchTxConfNotifications"
-	notifCh := make(chan confNotifMsg)
-	go func(ch chan confNotifMsg) {
-		resp, err := notifSource()
-		ch <- confNotifMsg{
-			resp: resp,
-			err:  err,
-		}
-
-		if resp != nil {
-			log.Tracef("Got notification with %d confirmation(s)",
-				len(resp.Confirmations))
-			for _, conf := range resp.Confirmations {
-				txHash, err := chainhash.NewHash(conf.TxHash)
-				if err != nil {
-					log.Tracef("Could not decode txhash: %b", conf.TxHash)
-				} else {
-					log.Tracef("    tx=%s, confs=%d, height=%d",
-						txHash, conf.Confirmations, conf.BlockHeight)
-				}
-			}
-		}
-	}(notifCh)
-
-	select {
-	case <-ctx.Done():
-		log.Tracef("%s: context cancelled fetching tx confirmation "+
-			"notifications", funcName)
-		return nil, errs.ContextCancelled
-	case notif := <-notifCh:
-		close(notifCh)
-		if notif.err != nil {
-			desc := fmt.Sprintf("%s: unable to fetch tx confirmation "+
-				"notifications, %s", funcName, notif.err)
-			return nil, errs.PoolError(errs.TxConf, desc)
-		}
-		return notif.resp, nil
-	}
-}
-
 // confirmCoinbases ensures the coinbases referenced by the provided
```

Comment:
I think this comment needs updating/expanding. Please explain what confirmCoinbases is doing, how it is doing it, and why it is necessary.

Reply:
Sure thing, updated.